### PR TITLE
feature/user-notifications

### PR DESCRIPTION
**Pull request by MennaAkram**

### CODE REVIEW

File: user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+fun NotificationsContent(
+    state: NotificationsUiState,
+    listener: NotificationsInteractionListener,
+) {
+    AppBarScaffold {
+        Column(
+            modifier = Modifier
+                .fillMaxSize(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(24.dp)
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt:54]: material theme

File: user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                state = state.emptyNotificationsPlaceHolder(),
+                image = R.drawable.placeholder_wish_list,
+                title = stringResource(R.string.your_notifications_is_empty),
+                subtitle = stringResource(R.string.you_ll_receive_a_notification_after_placing_your_order),
+                onClickDiscoverMarkets = {},
+                visibility = false
+            )
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.spacedBy(32.dp),
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt:71]: here too

File: user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    } else {
+                        black37
+                    },
+                    text = stringResource(R.string.delivery),
+                    onClickState = listener::onGetDeliveryNotifications
+                )
+            }
+                LazyColumn(
+                    modifier = Modifier
+                        .clip(RoundedCornerShape(24.dp))
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsScreen.kt:108]: here too

File: user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    private fun onError(error: ErrorHandler) {
+        _state.update { it.copy(isLoading = false, error = error) }
+        if (error is ErrorHandler.NoConnection) {
+            _state.update { it.copy(isError = true) }
+        }
+    }
+
+    private fun convertDate(date: String): String {
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/NotificationsViewModel.kt:75]: i dont know wich is better but i think this function should be in state not here

File: user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/composable/NotificationCard.kt [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.dp
+import org.the_chance.honymart.ui.theme.black60
+import org.the_chance.honymart.ui.theme.black87
+import org.the_chance.honymart.ui.theme.dimens
+import org.the_chance.honymart.ui.theme.white
+import org.the_chance.honymart.ui.theme.white200
+
+@Composable
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/notifications/composable/NotificationCard.kt:27]: just put all to material theme

File: user/src/main/java/org/the_chance/honeymart/ui/navigation/HomeNavGraph.kt [hunk 6]
```diff
@@ -4,15 +4,17 @@ import androidx.navigation.NavGraphBuilder
...
 import org.the_chance.honeymart.ui.feature.market.homeRoute
+import org.the_chance.honeymart.ui.feature.notifications.notificationsRoute
 import org.the_chance.honeymart.ui.feature.product.productRoute
 import org.the_chance.honeymart.ui.feature.product_details.productDetailsRoute
 import org.the_chance.honeymart.ui.feature.search.searchRoute
 
 fun NavGraphBuilder.homeNavGraph() {
     navigation(
-        startDestination = Screen.HomeScreen.route,
+        startDestination = Screen.NotificationsScreen.route,
```
Comment (AbdelrahmanMohamed-m) [inline user/src/main/java/org/the_chance/honeymart/ui/navigation/HomeNavGraph.kt:14]: remove this please

**Review by Abdelrahman-Sedky — APPROVED** [review]

**Review by TarekIdrees — CHANGES_REQUESTED** [review]
Is the notification sent as a flow? I mean in case the user uses the application and the notification received during the use of the application. Because I see that every function triggered from the service until the view model is used as a suspend function, not a flow.

**Review by Eng-Light — APPROVED** [review]

**Review by TarekIdrees — APPROVED** [review]

**Pull request merged**